spec/models/firm_factory_spec.rb

```diff
@@ -0,0 +1,189 @@
+RSpec.describe 'Firm factory' do
+  subject do
+    begin
+      FactoryGirl.create(factory)
+    rescue ActiveRecord::RecordInvalid
```

Comment:
the comment makes it better. i guess i don't know enough but I would of thought that if the create fails that would sufficient and there would be no need to initialise the object and test that instead. if your apps behaving like that under the hood then i guess it's fine